force-app/main/test/IBMVisualRecognitionV4FTests.cls

```diff
@@ -0,0 +1,179 @@
+public with sharing class IBMVisualRecognitionV4FTests {
+  private static String VERSION = '2017-09-21';
+  private static String URL = 'https://gateway.watsonplatform.net/visual-recognition/api';
+  private static String PRE_MADE_COLLECTION_ID = 'd31d6534-3458-40c4-b6de-2185a5f3cbe4';
+  private static String DOG_IMAGE_URL = 'https://upload.wikimedia.org/wikipedia/commons/thumb/4/47/American_Eskimo_Dog.jpg/1280px-American_Eskimo_Dog.jpg';
+  private static String CAT_IMAGE_URL = 'https://upload.wikimedia.org/wikipedia/commons/thumb/4/4f/Felis_silvestris_catus_lying_on_rice_straw.jpg/1280px-Felis_silvestris_catus_lying_on_rice_straw.jpg';
+
+   public static void runAllTests(String apiKey) {
+    IBMWatsonAuthenticator authenticator = new IBMWatsonIAMAuthenticator(apiKey);
+    IBMVisualRecognitionV4 service = new IBMVisualRecognitionV4(VERSION, authenticator);
+    service.setServiceURL(URL);
+
+    testAnalyze(service);
+    testCollectionOperations(service);
+    testImageOperations(service);
+    testDeleteUserData(service);
+  }
+
+  public static void testAnalyze(IBMVisualRecognitionV4 service) {
+    IBMVisualRecognitionV4Models.AnalyzeOptions options = new IBMVisualRecognitionV4Models.AnalyzeOptionsBuilder()
+      .addFeature('objects')
```

Comment:
Is collectionId required here? looks like we have `PRE_MADE_COLLECTION_ID`

Reply:
Good call. Didn't catch it since we're not running these yet 😛